### ISSUE TITLE
[BUG] Logic Pro crash when changing signal chain order

### ISSUE THREAD

**Issue by mparker** [sdatkinson/NeuralAmpModelerPlugin#205]
**Thanks for taking the time to write a bug report!** Use the following prompts to help you describe what's going on.  The more info you provide, the easier it'll be to understand how to fix it for you without having to come back and ask you questions.

**Describe the bug**
Logic Pro crashes when I changed the order of a different plugin in the signal chain while NAM is in it.

**To Reproduce**
1. Add NAM to the signal chain, along with other plugins (in my case it was a compressor, pitch shifter)
2. Change the plugin order (I moved the compressor above NAM)

**Desktop (please complete the following information):**
 - macOS 13.3.1
 - Plugin version 0.7.1
 - AU
 - Log Pro X

```
Process:               Logic Pro X [30570]
Path:                  /Applications/Logic Pro X.app/Contents/MacOS/Logic Pro X
Identifier:            com.apple.logic10
Version:               10.7.7 (5762)
Build Info:            MALogic-5762000000000000~2 (1A85)
App Item ID:           634148309
App External ID:       854029738
Code Type:             X86-64 (Native)
Parent Process:        launchd [1]
User ID:               502

Date/Time:             2023-04-17 20:55:52.0943 -0600
OS Version:            macOS 13.3.1 (22E261)
Report Version:        12
Bridge OS Version:     3.0 (14Y910)
Anonymous UUID:        3FA3F436-08C9-4D87-CE0D-E59C42AAF8CD

Sleep/Wake UUID:       FEE73997-0F73-46DF-86E0-3EF34F594ADC

Time Awake Since Boot: 150000 seconds
Time Since Wake:       48790 seconds

System Integrity Protection: enabled

Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0x000000000000013c
Exception Codes:       0x0000000000000001, 0x000000000000013c

Termination Reason:    Namespace SIGNAL, Code 11 Segmentation fault: 11
Terminating Process:   exc handler [30570]

VM Region Info: 0x13c is not in any region.  Bytes before following region: 140737486937796
      REGION TYPE                    START - END         [ VSIZE] PRT/MAX SHRMOD  REGION DETAIL
      UNUSED SPACE AT START
--->  
      shared memory            7fffffea6000-7fffffea7000 [    4K] r-x/r-x SM=SHM  

Thread 0 Crashed::  Dispatch queue: com.apple.main-thread
0   NeuralAmpModeler              	       0x14b017a0f 0x14b006000 + 72207
1   NeuralAmpModeler              	       0x14b0be6ce 0x14b006000 + 755406
2   NeuralAmpModeler              	       0x14b068e92 0x14b006000 + 405138
3   NeuralAmpModeler              	       0x14b0b1532 0x14b006000 + 701746
4   NeuralAmpModeler              	       0x14b06b2af 0x14b006000 + 414383
5   MAAudioUnitSupport            	       0x103ba2349 0x103b8d000 + 86857
6   MAAudioEngine                 	       0x1066513a6 CMDAudioUnit::MakeEditorView(long, TGroupID) + 294
7   MAAudioEngine                 	       0x1064a073d MDMultiMonoPlug::MakeEditorView(long, TGroupID) + 509
8   Logic Pro X                   	       0x102a25b15 0x101418000 + 23124757
9   Logic Pro X                   	       0x1025c93b3 0x101418000 + 18551731
10  Logic Pro X                   	       0x1025c8b46 0x101418000 + 18549574
11  Logic Pro X                   	       0x1025ca3c0 0x101418000 + 18555840
12  Logic Pro X                   	       0x1025cdc55 0x101418000 + 18570325
13  Logic Pro X                   	       0x10169dc56 0x101418000 + 2645078
14  Logic Pro X                   	       0x102d78a54 0x101418000 + 26610260
15  Logic Pro X                   	       0x101af8706 0x101418000 + 7210758
...
      0x14b006000 -        0x14b0d5fff com.StevenAtkinson.audiounit.NeuralAmpModeler (0.7.1) <9b9e562b-f1c6-3dbd-a922-cbe7a8dfa485> /Library/Audio/Plug-Ins/Components/NeuralAmpModeler.component/Contents/MacOS/NeuralAmpModeler
```

**Additional context**

**Comment by sdatkinson** [comment]
Was likely solved with #298

**Comment by mparker** [comment]
confirmed resolved in 0.7.5
